seed-job/template.yaml

```diff
@@ -55,76 +55,89 @@ objects:
                     def image_name = "${APP_ID}/${JOB_ID}:${DESIRED_TAG}"
                     def image_name_with_registry = "${REGISTRY_URL}/${image_name}"
                     def daemonset_name = "scan-data_scan-data"
-                    stage('Checkout Source') {
-                        checkout([
-                            $class: 'GitSCM',
-                            branches: [[name: "${GIT_BRANCH}"]],
-                            doGenerateSubmoduleConfigurations: false,
-                            extensions: [
-                                [$class: 'RelativeTargetDirectory', relativeTargetDir: "${PIPELINE_NAME}"],
-                                [$class: 'CloneOption', depth: 1, noTags: false, reference: '', shallow: true, timeout: 10]
-                            ],
-                            submoduleCfg: [],
-                            userRemoteConfigs: [
-                                [url: "${GIT_URL}"]
-                            ]
-                        ])
-                    }
-                    stage('Prebuild source repo'){
-                        if("${PRE_BUILD_SCRIPT}" != "None")
-                        {
-                           dir("${PIPELINE_NAME}/${PRE_BUILD_CONTEXT}"){
-                              sh "sh ${PRE_BUILD_SCRIPT}"
-                           }
-                        }
-                        else
-                        {
-                            echo "Prebuild is not requested"
+                    def success = false
+                    try {
```

Comment:
try block encapsulate all stages, do we really want to make stages like `linting` and `scanning` a mandatory stage which can fail the builds?

Reply:
Linting and scanning are a part of the basic features that we provide through our service. Hence they have been encapsulated here.